.circleci/config.yml

```diff
@@ -0,0 +1,44 @@
+version: 2.1
+
+commands:
+  # Checkout repo, restore cache and update Cargo.lock
+  common_init:
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "eg-0.6/Cargo.lock" }}-{{ checksum "eg-next/Cargo.lock" }}
```

Comment:
Do we want to store the `Cargo.lock` for `eg-next` in the repo? I think this would cause the git dependency to be fixed to a single release until `cargo update` is called, which wouldn't be the expected behavior for a daily build of this repo.

Reply:
That's a good point. I've left `Cargo.lock` both tracked by git and in this cache key, but I'm running `cargo update` before every build in CI. This invalidates the cache somewhat (until `cargo update` is performed by a developer and committed), but keeps things consistent between the other folders and this one.

I can add `eg-next/Cargo.lock` to `.gitignore` if you think that's a better solution though. I'm just wondering if it will hinder local dev much or not.

Reply:
Never mind! That approach causes more problems than it solves... I've gitignored  `eg-next/Cargo.lock`. Let's see if that works...